### PR TITLE
Refactor userData (remove KeyValueProperty)

### PR DESCRIPTION
**Pull request by RubenMattan**
```python
def migrate_userdata():
  from rogerthat.migrations.migrate_user_data import job
  
  job()
```

### CODE REVIEW

File: src/rogerthat/models/__init__.py [hunk 1]
```diff
@@ -474,8 +474,8 @@ def create(cls, app_user):
 
 
 class UserData(db.Model):
-    data = db.TextProperty()  # deprecated, lazily migrated to userData when putting user_data
-    userData = KeyValueProperty()
+    data = db.TextProperty() #reactived for ndb migration # deprecated, lazily migrated to userData when putting user_data
```
Comment (lucasvanhalst) [inline src/rogerthat/models/__init__.py:477]: Just remove the comment, not really needed

File: src/rogerthat/bizz/service/__init__.py [hunk 2]
```diff
@@ -2749,33 +2749,29 @@ def trans(updated_json_dict):
...
-            elif user_data.userData is None:
-                user_data.userData = KVStore(user_data_key)
-                if user_data.data:
-                    full_json_dict = json.loads(user_data.data)
+            elif user_data.data is None:
+                if user_data.userData:
+                    full_json_dict = user_data.userData.to_json_dict()
                     full_json_dict.update(updated_json_dict)
-                user_data.data = None
+                    user_data.userData.clear()
```
Comment (lucasvanhalst) [inline src/rogerthat/bizz/service/__init__.py:2758]: Is this still needed? I thought the migration removes this property value

File: src/rogerthat/bizz/service/__init__.py [hunk 3]
```diff
@@ -2749,33 +2749,29 @@ def trans(updated_json_dict):
...
             else:
-                full_json_dict = user_data.userData.to_json_dict()
+                full_json_dict = json.loads(user_data.data)
                 full_json_dict.update(updated_json_dict)
+                
         else:
             user_data = UserData(key=user_data_key,
                                  data=None,
-                                 userData=KVStore(user_data_key))
+                                 userData=None)
```
Comment (lucasvanhalst) [inline src/rogerthat/bizz/service/__init__.py:2766]: Same as above, still needed? If the model doesn't exist yet there shouldn't be a reason to set this property

File: src/rogerthat/bizz/user/export.py [hunk 4]
```diff
@@ -151,10 +151,10 @@ def run(self, result_path, human_user_email, app_id, service_identity_email):
         si = get_service_identity(service_identity_user)
         ud = UserData.get(UserData.createKey(app_user, add_slash_default(service_identity_user)))
         if ud:
-            if ud.userData:
-                user_data = ud.userData.to_json_dict()
-            else:
+            if ud.data:
                 user_data = json.loads(ud.data)
+            else:
+                user_data = ud.userData.to_json_dict()
```
Comment (lucasvanhalst) [inline src/rogerthat/bizz/user/export.py:157]: i'd remove this as user exports don't run very often

File: src/rogerthat/migrations/migrate_user_data.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    def trans():
+        user_data = db.get(user_data_key)
+        if user_data.data:
+            return
+        if not user_data.userData:
+            logging.debug("data empty app_user:%s service_identity_user:%s" % (user_data.app_user, user_data.service_identity_user))
+            raise Exception("migrate_user_data failed")
+        
+        data_dict = user_data.userData.to_json_dict()
+        user_data.userData.clear()
```
Comment (lucasvanhalst) [inline src/rogerthat/migrations/migrate_user_data.py:47]: maybe add  `user_data.userData = None` as well

File: src/rogerthat/bizz/service/__init__.py (hunk 2 above)
Comment (RubenMattan) [reply src/rogerthat/bizz/service/__init__.py:2758]: this way it still works before migration or during the migration

File: src/rogerthat/bizz/user/export.py (hunk 4 above)
Comment (RubenMattan) [reply src/rogerthat/bizz/user/export.py:157]: will be removed after the migration

File: src/rogerthat/migrations/migrate_user_data.py (hunk 5 above)
Comment (RubenMattan) [reply src/rogerthat/migrations/migrate_user_data.py:47]: then the buckets aren't cleared

**Pull request merged**